boot-mongodb-elasticsearch/src/test/java/com/example/mongoes/web/controller/SearchControllerIntTest.java

```diff
@@ -1,18 +1,346 @@
+                                        .queryParam("limit", 10)
+                                        .build())
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus()
+                .isOk()
+                .expectBody(new ParameterizedTypeReference<SearchPage<Restaurant>>() {});
+        //                .value(page -> assertThat(page.totalElements()).isEqualTo(1L));
+    }
+
+    @Test
+    @Disabled
+    void searchRestaurantIdRange_ShouldReturnSearchPage() {
+
+        webTestClient
+                .get()
+                .uri(
+                        uriBuilder ->
+                                uriBuilder
+                                        .path("/search/restaurant/range")
+                                        .queryParam("lowerLimit", 1)
+                                        .queryParam("upperLimit", 100)
+                                        .build())
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus()
+                .isOk()
+                .expectBody(new ParameterizedTypeReference<SearchPage<Restaurant>>() {});
+        //                .value(page -> assertThat(page.totalElements()).isEqualTo(1L));
+    }
+
+    @Test
+    @Disabled
+    void searchDateRange_ShouldReturnSearchPage() {
+
+        webTestClient
+                .get()
+                .uri(
+                        uriBuilder ->
+                                uriBuilder
+                                        .path("/search/date/range")
+                                        .queryParam("fromDate", "2024-01-01")
+                                        .queryParam("toDate", "2024-12-31")
+                                        .queryParam("offset", 0)
+                                        .queryParam("limit", 10)
+                                        .build())
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus()
+                .isOk()
+                .expectBody(new ParameterizedTypeReference<SearchPage<Restaurant>>() {});
+        //                .value(page -> assertThat(page.totalElements()).isEqualTo(1L));
+    }
+
+    @Test
+    @Disabled
+    void aggregateSearch_ShouldReturnAggregationResponse() {
+
+        webTestClient
+                .get()
+                .uri(
+                        uriBuilder ->
+                                uriBuilder
+                                        .path("/search/aggregate")
+                                        .queryParam("searchKeyword", "garden")
+                                        .queryParam("fieldNames", "name", "cuisine")
+                                        .build())
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus()
+                .isOk()
+                .expectBody(AggregationSearchResponse.class);
```

Comment:
@coderabbitai, when running from main program in swagger it is rendering response but when running from test it is throwing below error


`org.springframework.core.codec.CodecException: Type definition error: [simple type, class org.springframework.data.domain.Pageable]` , please help me to fix this

Reply:
@coderabbitai, when used below curl, it is responding correctly but when trying using `WebTestClient` it is not working and giving above error. How to fix this?

```curl
curl -X 'GET' \
  'http://localhost:8080/search/aggregate?searchKeyword=Pizza&fieldNames=restaurant_name&fieldNames=borough&fieldNames=cuisine&limit=15&offset=0&sortOrder=DESC&sortFields=restaurant_id' \
  -H 'accept: */*'
```

Reply:
@coderabbitai, I have dig further into the issue and found that we need to manually add codec to webTestClient as per [link](https://stackoverflow.com/questions/51513023/manually-configure-jackson-module-for-spring-webtestclient) , can you please help me to add the same.

Reply:
@coderabbitai, still getting error as `org.springframework.core.codec.CodecException: Type definition error: [simple type, class org.springframework.data.domain.Pageable] `  with above change. 

can we use any other alternative to `WebTestClient`

Reply:
@coderabbitai, above is also not working. Can you please help me to migrate it to restassured  and verify content there.

Reply:
@coderabbitai, how to capture AggregationSearchResponse from above junit. As response of the endpoint is AggregationSearchResponse